docs/terraform-provider.md

```diff
@@ -0,0 +1,16 @@
+# Auth0 Terraform Provider
```

Comment:
Perhaps we can squeeze in the provider docs somewhere in here as well: https://registry.terraform.io/providers/auth0/auth0/latest